### DISCUSSION TITLE
Best approach for rapid data change

### DISCUSSION THREAD

**Discussion by karllapuz** [visgl/deck.gl#7315] [Q&A]
I'm rendering a simple line layer with data points that rapidly change due to frequent frame change and I'm running into an issue where I'm assuming the previous line layer doesn't get unmounted as quickly as the new data comes in, which results in lines doubling. What are some best practices when handling layer data changing quickly? If my assumption is correct, how do I alleviate this issue or does deckgl handle all layer disposal under the hood when data prop changes? 

Thanks!

**Comment by Pessimistress** [comment]
If you are "replacing" old data, then you should update an existing layer's data prop to the new value. Finalizing and initializing layers are very expensive.
